test/ssm_document/helper/helper.go

```diff
@@ -0,0 +1,85 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+package helper
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go-v2/service/ssm"
+	"github.com/aws/aws-sdk-go-v2/service/ssm/types"
+
+	"github.com/aws/amazon-cloudwatch-agent-test/util/awsservice"
+)
+
+type AgentStatus struct {
+	Status       string `json:"status"`
+	ConfigStatus string `json:"configstatus"`
+	Version      string `json:"version"`
+	StartTime    string `json:"starttime"`
+}
+
+func RunSSMAction(documentName string, instanceIds []string, params map[string][]string, actionName, expectedStatus, expectedConfigStatus string) error {
+	log.Printf("Testing %s action", actionName)
+
+	out, err := awsservice.RunSSMDocument(documentName, instanceIds, params)
+	if err != nil {
+		return fmt.Errorf("%s action failed: %v", actionName, err)
+	}
+
+	if err := VerifyAgentActionError(out, instanceIds[0], documentName, expectedStatus, expectedConfigStatus); err != nil {
+		return fmt.Errorf("%s verification failed: %v", actionName, err)
+	}
+
+	log.Printf("%s action completed successfully", actionName)
+	return nil
+}
+
+func VerifyAgentActionError(out *ssm.SendCommandOutput, instanceId, documentName, expectedAgentStatus string, expectedConfigStatus string) error {
```

Comment:
nit: probably don't need the "Error" in the func name.
```suggestion
func VerifyAgentAction(out *ssm.SendCommandOutput, instanceId, documentName, expectedAgentStatus string, expectedConfigStatus string) error {
```